File: Testing/Code for Testing/Early Basic Syntax Testing/yield/test_yield.js

function* countdown(n) {
    while ((n > 0)) {
        yield n;
        n -= 1;
    }
}

function* doubles(n) {
    let i = 1;
    while ((i <= n)) {
        yield (i * 2);
        i += 1;
    }
}

function main() {
    for (const val of countdown(5)) {
        console.log(val);
    }
    for (const d of doubles(4)) {
        console.log(d);
    }
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();

File: Testing/Code for Testing/Early Basic Syntax Testing/yield/test_yield.py
from dataclasses import dataclass, field

def _ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class _ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def countdown(n):
    while (n > 0):
        yield n
        n -= 1

def doubles(n):
    i = 1
    while (i <= n):
        yield (i * 2)
        i += 1

def main():
    for val in countdown(5):
        print(val)
    for d in doubles(4):
        print(d)
    return 0


if __name__ == "__main__":
    main()
